cns/nmagent/client.go

```diff
@@ -65,13 +71,30 @@ func NewClient(url string) (*Client, error) {
 }
 
 // JoinNetwork joins the given network
-func JoinNetwork(networkID, joinNetworkURL string) (*http.Response, error) {
+func JoinNetwork(networkID string) (*http.Response, error) {
 	logger.Printf("[NMAgentClient] JoinNetwork: %s", networkID)
 
 	// Empty body is required as wireserver cannot handle a post without the body.
 	var body bytes.Buffer
 	json.NewEncoder(&body).Encode("")
-	response, err := common.GetHttpClient().Post(joinNetworkURL, "application/json", &body)
+
+	joinNetworkTypeValue := fmt.Sprintf(
+		JoinNetworkURLFmt,
+		networkID)
+
+	joinNetworkURL := url.URL{
+		Host:   WireserverIP,
+		Path:   WireServerPath,
+		Scheme: WireServerScheme,
+	}
+
+	queryString := joinNetworkURL.Query()
+	queryString.Set("type", joinNetworkTypeValue)
+	queryString.Set("comp", "nmagent")
+
+	joinNetworkURL.RawQuery = queryString.Encode()
+
+	response, err := common.GetHttpClient().Post(joinNetworkURL.String(), "application/json", &body)
```

Comment:
since there are a few instances of this linting error, I'd suggest a small utility function:
```go
func PostCtx(ctx context.Context, cl *http.Client, url string, contentType string, body io.Reader) (*http.Response, error) {
	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, body)
	if err != nil {
		return nil, fmt.Errorf("could not create POST request: %w", err)
	}

	req.Header.Set("Content-Type", contentType)
	return cl.Do(req)
}
```
such that this line
```go
response, err := common.GetHttpClient().Post(joinNetworkURL.String(), "application/json", &body)
```
can be replaced with
```go
response, err := common.PostCtx(context.TODO(), common.GetHttpClient(), joinNetworkURL.String(), "application/json", &body)
```
which I think should address this.